src/main/java/org/testng/SuiteRunner.java

```diff
@@ -220,9 +247,17 @@ public int compare(XmlTest arg0, XmlTest arg1) {
       }
     });
 
+    boolean isNewVariant = (iTestRunnerFactory instanceof ITestRunnerFactory2);
```

Comment:
And, in that case, `ITestRunnerFactory2` doesn't need to extend `ITestRunnerFactory`.